cpp/src/community/k_truss_impl.cuh

```diff
@@ -0,0 +1,920 @@
+std::tuple<rmm::device_uvector<vertex_t>,
+           rmm::device_uvector<vertex_t>,
+           std::optional<rmm::device_uvector<weight_t>>>
+k_truss(raft::handle_t const& handle,
+        graph_view_t<vertex_t, edge_t, false, multi_gpu> const& graph_view,
+        std::optional<edge_property_view_t<edge_t, weight_t const*>> edge_weight_view,
+        edge_t k,
+        bool do_expensive_check)
+{
+  // 1. Check input arguments.
+
+  CUGRAPH_EXPECTS(!graph_view.has_edge_mask(), "unimplemented.");
+
+  CUGRAPH_EXPECTS(graph_view.is_symmetric(),
+                  "Invalid input arguments: K-truss currently supports undirected graphs only.");
+  CUGRAPH_EXPECTS(!graph_view.is_multigraph(),
+                  "Invalid input arguments: K-truss currently does not support multi-graphs.");
+
+  if (do_expensive_check) {
+    // nothing to do
+  }
+
+  std::optional<graph_t<vertex_t, edge_t, false, multi_gpu>> modified_graph{std::nullopt};
+  std::optional<graph_view_t<vertex_t, edge_t, false, multi_gpu>> modified_graph_view{std::nullopt};
+  std::optional<rmm::device_uvector<vertex_t>> renumber_map{std::nullopt};
+  std::optional<edge_property_t<graph_view_t<vertex_t, edge_t, false, multi_gpu>, weight_t>>
+    edge_weight{std::nullopt};
+
+  if (graph_view.count_self_loops(handle) > edge_t{0}) {
+    auto [srcs, dsts] = extract_transform_e(handle,
+                                            graph_view,
+                                            edge_src_dummy_property_t{}.view(),
+                                            edge_dst_dummy_property_t{}.view(),
+                                            edge_dummy_property_t{}.view(),
+                                            exclude_self_loop_t<vertex_t>{});
+
+    if constexpr (multi_gpu) {
+      std::tie(srcs, dsts, std::ignore, std::ignore, std::ignore) =
+        detail::shuffle_ext_vertex_pairs_with_values_to_local_gpu_by_edge_partitioning<vertex_t,
+                                                                                       edge_t,
+                                                                                       weight_t,
+                                                                                       int32_t>(
+          handle, std::move(srcs), std::move(dsts), std::nullopt, std::nullopt, std::nullopt);
+    }
+
+    std::tie(*modified_graph, std::ignore, std::ignore, std::ignore, renumber_map) =
+      create_graph_from_edgelist<vertex_t, edge_t, weight_t, edge_t, int32_t, false, multi_gpu>(
+        handle,
+        std::nullopt,
+        std::move(srcs),
+        std::move(dsts),
+        std::nullopt,
+        std::nullopt,
+        std::nullopt,
+        cugraph::graph_properties_t{true, graph_view.is_multigraph()},
+        true);
+
+    modified_graph_view = (*modified_graph).view();
+  }
+
+  // FIXME: Remove because it yields to incorrect results
+  // 3. Find (k+1)-core and exclude edges that do not belong to (k+1)-core
```

Comment:
(k+1)=>(k-1) and re-enable this. This shouldn't yield to incorrect results.

Reply:
I tried and it still yield incorrect results. I tried even `k-2`

Reply:
I added a `fixme` for this